addons/mail/models/res_users_settings.py

```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class ResUsersSettings(models.Model):
+    _name = 'res.users.settings'
+    _description = 'Mail User Settings'
+
+    user_id = fields.Many2one('res.users', string="User", required=True)
+    is_discuss_sidebar_category_channel_open = fields.Boolean(string="Is discuss sidebar category channel open?", default=True)
+    is_discuss_sidebar_category_chat_open = fields.Boolean(string="Is discuss sidebar category chat open?", default=True)
+
+    _sql_constraints = [
+        ('unique_user_id', 'UNIQUE(user_id)', 'One user should only have one mail user settings.')
+    ]
+
+    def _res_users_settings_format_fields(self):
+        return [
+            'id',
+            'is_discuss_sidebar_category_channel_open',
+            'is_discuss_sidebar_category_chat_open',
+            'user_id',
+        ]
+
+    def _res_users_settings_format(self):
```

Comment:
Not sure, why adding "_res_users_settings" prefix ? As anyway you are on res.users.settings model ... simply _format_for_ui or something like that ?

Reply:
It was inspired by `message_format`, `channel_info`, etc. They all have the model name in the method name.

Reply:
Mmh, actually message_format is a weird one. "message" is generally used for mail.thread methods (to avoid mixin conflicting with standard model code). However "format" is too generic ... well, anyway not critical for this PR. But somehow this could become _discuss_format, _format_for_ui, _format_web, ... ? Maybe post-v15 .